### PR TITLE
gh-127434: Fix iOS `xcrun --sdk` clang/ar scripts to allow arguments with spaces

### LINKED ISSUE #127434: iOS `clang` aliases crash with "invalid directory name" error if called with arguments having spaces
# Bug report

### Bug description:

While doing cross-compilation of Python packages for iOS calling [`arm64-apple-ios-clang`](https://github.com/python/cpython/blob/main/iOS/Resources/bin/arm64-apple-ios-clang) script (and other aliases as well) fails if arguments contain spaces, for example:

```
arm64-apple-ios-clang -DPACKAGE_TARNAME="aaa" -DPACKAGE_STRING="aaa 1.2.2" -c -o src/main.o src/main.c
```

gives `clang: invalid directory name: '1.2.2"'` error.

To fix that `$@` inside script should be wrapped with quotes: `"$@"`.

I'm going to make PR.
 

### CPython versions tested on:

3.12

### Operating systems tested on:

macOS

### Linked PRs
* gh-127575
* gh-127624

### PR DESCRIPTION
**Pull request by FeodorFitsner**
This PR fixes scripts `iOS/Resources/bin` directory to allow arguments with spaces.

Fixes #127434

* Issue: gh-127434

### CODE REVIEW

**Comment by FeodorFitsner** [comment]
cc @freakboy3742

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Thanks for the contribution - the code changes look fine; from a process perspective, all changes to Python [require a NEWS entry](https://devguide.python.org/core-developers/committing/#how-to-add-a-news-entry). Bevedere has provided a link to an online tool that can be used to add one, or you can set up the `blurb` tool in your own environment.

File: Misc/NEWS.d/next/Core_and_Builtins/2024-12-04-09-52-08.gh-issue-127434.RjkGT_.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix clang/ar scripts in `iOS/Resources/bin` directory to allow arguments
+with spaces.
```
Comment (freakboy3742) [inline Misc/NEWS.d/next/Core_and_Builtins/2024-12-04-09-52-08.gh-issue-127434.RjkGT_.rst]:
```suggestion
The iOS compiler shims can now accept arguments with spaces.
```

**Review by freakboy3742 — APPROVED** [review]
Thanks for the fix!

**Comment by freakboy3742** [comment]
!buildbot iOS

**Comment by freakboy3742** [comment]
I'm just waiting on an iOS buildbot job to 100% confirm that this works; for some reason, the buildbots doesn't seem to be starting jobs at present...

**Comment by freakboy3742** [comment]
The Fedora build failure seems to be a transient network failure, not related to this PR.

**Pull request merged**